test/unit/test_ibm_primitives_v2.py

```diff
@@ -141,22 +142,28 @@ def __new__(cls, *args, **kwargs):  # pylint: disable=unused-argument
 
         with patch("qiskit_ibm_runtime.base_primitive.QiskitRuntimeService", new=MockQRTService):
             inst = primitive(backend=backend_name)
-            self.assertIsNone(inst.session)
+            self.assertIsNone(inst.mode)
             inst.run(**get_primitive_inputs(inst))
             mock_service_inst.run.assert_called_once()
             runtime_options = mock_service_inst.run.call_args.kwargs["options"]
             self.assertEqual(runtime_options["backend"], mock_backend)
 
+            mock_service_inst.reset_mock()
+            str_mode_inst = primitive(mode=backend_name)
+            self.assertIsNone(str_mode_inst.mode)
+            inst.run(**get_primitive_inputs(str_mode_inst))
+            mock_service_inst.run.assert_called_once()
+            runtime_options = mock_service_inst.run.call_args.kwargs["options"]
+            self.assertEqual(runtime_options["backend"], mock_backend)
+
     @data(EstimatorV2, SamplerV2)
     def test_init_with_session_backend_str(self, primitive):
         """Test initializing a primitive with a backend name using session."""
         backend_name = "ibm_gotham"
 
         with patch("qiskit_ibm_runtime.base_primitive.QiskitRuntimeService"):
-            with self.assertRaises(ValueError) as exc:
-                inst = primitive(session=backend_name)
-                self.assertIsNone(inst.session)
-            self.assertIn("session must be of type Session or None", str(exc.exception))
+            inst = primitive(session=backend_name)
+            self.assertIsNone(inst.mode)
```

Comment:
I don't see the point of having this test now. It was added to ensure one couldn't specify a backend name as a session (which was allowed way back). Now that it no longer raises an error but violates type hint, I'm not sure what the test is trying to cover.